### PR TITLE
upgrade file format

### PR DESCRIPTION
**Pull request by abuts**
This solves #1656. It looks like it also addresses Re #1498 and Re #1195

### CODE REVIEW

File: herbert_core/configuration/log_config.m [hunk 1]
```diff
@@ -60,8 +60,8 @@
     end
 
     properties(Access=protected, Hidden=true)
-        info_log_print_time_ = 30 % normally info log should be printed
-        %                         % every 30 secodd
+        info_log_print_time_ = 300 % normally info log should be printed
```
Comment (oerc0122) [inline herbert_core/configuration/log_config.m:63]: Why? I understand that this comment wasn't added this PR, but why should the log be printed every X timeperiod?

File: horace_core/admin/upgrade_file_format.m [hunk 2]
```diff
@@ -46,20 +41,24 @@
...
             if nargout > 0
-                sqw_list{i} = sqw(ld_new,'file_backed',true);
+                sqw_list{i} = finalize_alignment(ld_new);
+            else
+                finalize_alignment(ld_new);   % Will do nothing if the file is not aligned && ranges are valid
             end
-            ld_new.delete();
+        elseif nargout>0
+            sqw_list{i} = sqw(ld_new,'file_backed',true);
         end
```
Comment (oerc0122) [inline horace_core/admin/upgrade_file_format.m:60]: Would prefer like elements grouped together
```suggestion
        if nargout > 0
            if upgrade_ranges %
                sqw_list{i} = finalize_alignment(ld_new);
            else
                sqw_list{i} = sqw(ld_new,'file_backed',true);
            end
        else
            finalize_alignment(ld_new);   % Will do nothing if the file is not aligned && ranges are valid
        end
```

File: horace_core/sqw/PixelData/@PixelDataBase/finalize_alignment.m [hunk 3]
```diff
@@ -11,7 +11,7 @@
     outfile = '';
 end
 targ_obj = copy(obj);
-if ~targ_obj.is_misaligned
+if ~targ_obj.is_misaligned && targ_obj.is_range_valid
```
Comment (oerc0122) [inline horace_core/sqw/PixelData/@PixelDataBase/finalize_alignment.m:14]: This seems to be a repeated condition with a defined meaning. Possibly want a getter for this combination with a clear name.

File: horace_core/sqw/PixelData/@pix_metadata/pix_metadata.m [hunk 4]
```diff
@@ -58,6 +58,20 @@
...
+            % input is valid, i.e. not equal to empty range;
+            if nargin == 1 % check the whole range
+                idx = PixelDataBase.FIELD_INDEX_MAP('all');
+            elseif iscell(fld)
+                idx = cellfun(@(fl)PixelDataBase.FIELD_INDEX_MAP(fl),fld);
+            else
+                idx = PixelDataBase.FIELD_INDEX_MAP(fld);
+            end
+            invalid = obj.data_range_(:,idx) == PixelDataBase.EMPTY_RANGE(:,idx);
+            is = ~any(invalid(:));
```
Comment (oerc0122) [inline horace_core/sqw/PixelData/@pix_metadata/pix_metadata.m:72]: Minor thing and subjective, but given the function is in the positive would probably find it easier to rationalise if this were in the positive too:
```suggestion
            valid = obj.data_range_(:,idx) ~= PixelDataBase.EMPTY_RANGE(:,idx);
            is = all(valid(:));
```

**Review by oerc0122 — CHANGES_REQUESTED** [review]

File: _test/test_admin/test_upgrade_file_format.m [hunk 5]
```diff
@@ -68,7 +65,9 @@ function test_upgrade_single_sqw_filebacked_upgrade_range_no_warning(obj)
             w2  = sqw(targ_f{1});
 
             [~,e] = lastwarn;
-            assertEqual(e,'TESTS:my_warning');
+            % second warning comes on windows as memmapfile is locked
+            % despite beeing deleted.
+            assertTrue(isequal(e,'TESTS:my_warning')||isequal(e,'MATLAB:DELETE:Permission'));
```
[the remark this answers is not in the document]
Comment (abuts) [reply _test/test_admin/test_upgrade_file_format.m:70]: no masking -- it has been deleted but warning was issued.

File: _test/test_sqw_pixels/test_PixelAlignment.m [hunk 6]
```diff
@@ -39,6 +39,28 @@ function test_save_filebacked_applies_alignment_on_sqw(~)
...
+            pdm = PixelDataFileBacked(pix_data);
+            
+            assertFalse(pdm.is_misaligned);
+            assertTrue(pdm.is_range_valid);
+
+            ref_range = pdm.data_range;
+            pdm.data_range = PixelDataBase.EMPTY_RANGE;
+            assertFalse(pdm.is_range_valid);
+
+            pdm = pdm.finalize_alignment();
```
[the remark this answers is not in the document]
Comment (abuts) [reply _test/test_sqw_pixels/test_PixelAlignment.m:57]: well, yes, but I would not doing it now.

File: herbert_core/configuration/log_config.m (hunk 1 above)
Comment (abuts) [reply herbert_core/configuration/log_config.m:63]: I was working with alignment and found this issue. This number controls log string (`printed N#out of #F`). Dots are printed with every page

File: horace_core/admin/upgrade_file_format.m (hunk 2 above)
Comment (abuts) [reply horace_core/admin/upgrade_file_format.m:60]: I am very reluctant to change order of operations here on a whim. The operations have costs there and in suggested case you will read metadata regardless of your intentions on aligning them. This change needs much more thinking than I am ready to spend on it

File: horace_core/sqw/PixelData/@PixelDataBase/finalize_alignment.m (hunk 3 above)
Comment (abuts) [reply horace_core/sqw/PixelData/@PixelDataBase/finalize_alignment.m:14]: they appear in various independent places. Its not so big and complex thing, to define new API for it. The code is much clearer this way.

**Pull request merged**